Sources/ATTNSDK.m

```diff
@@ -100,9 +101,10 @@ - (void)trigger:(UIView *)theView handler:(ATTNCreativeTriggerCompletionHandler)
 
   [[wkWebViewConfiguration userContentController] addScriptMessageHandler:self name:@"log"];
 
-  NSString *userScriptWithEventListener = @"window.addEventListener('message', (event) => {if (event.data && event.data.__attentive && event.data.__attentive.action === 'CLOSE') {window.webkit.messageHandlers.log.postMessage(event.data.__attentive.action);}}, false);";
+  NSString *userScriptWithEventListener = @"window.addEventListener('message', (event) => {if (event.data && event.data.__attentive) {window.webkit.messageHandlers.log.postMessage(event.data.__attentive.action);}}, false);window.addEventListener('visibilitychange', (event) => {window.webkit.messageHandlers.log.postMessage(`Document Hidden: ${document.hidden}`);}, false);";
```

Comment:
I suggest moving the message `Document Hidden: ${document.hidden}` to a variable/function that we can template. The message as it stands is a little too "rich" in my opinion (various capitalizations, punctuation, spacing, etc) to be manually matched against down below. Maybe generate the string using a function like `generateVisibilityEvent(visibility: boolean): string`

Reply:
good point, will update